subprojects/workers/src/main/java/org/gradle/process/internal/daemon/WorkerDaemonClientsManager.java

```diff
@@ -79,4 +80,20 @@ public void stop() {
             allClients.clear();
         }
     }
+
+    /**
+     * Select idle daemon clients to stop.
+     *
+     * @param selectionFunction Gets all idle daemon clients, daemons of returned clients are stopped
+     */
+    public void selectIdleClientsToStop(Transformer<List<WorkerDaemonClient>, List<WorkerDaemonClient>> selectionFunction) {
+        synchronized (lock) {
+            List<WorkerDaemonClient> clientsToStop = selectionFunction.transform(new ArrayList<WorkerDaemonClient>(idleClients));
```

Comment:
I've implemented least frequently used for now, but will do some testing with both to see which seems to behave better.